core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -313,8 +316,8 @@ class HttpGoogleServicesDAO(
   override def deleteBucket(bucketName: String, monitorRef: ActorRef): Future[Unit] = {
     val buckets = getStorage(getBucketServiceAccountCredential).buckets
     val deleter = buckets.delete(bucketName)
-    retryWithRecoverWhen500orGoogleError(() => {
-      executeGoogleRequest(deleter)
+    retryWithRecoverWhen500orGoogleError(count => {
+      executeGoogleRequest(deleter)(GoogleInstrumentedService.Buckets, count)
```

Comment:
See `GoogleInstrumented.scala` and `GoogleInstrumentedServiceMapper.scala`. Basically I try to infer the `GoogleInstrumentedService` enum based on the type parameter of `AbstractGoogleClientRequest[T]` (it's like they want us to make a typeclass). Most of the time this is possible because there's a one-to-one mapping between type parameter and service -- e.g. `AbstractGoogleClientRequest[com.google.api.services.storage.model.Bucket]` is always GoogleBuckets request. However sometimes the type is ambiguous -- for example most delete requests are a `AbstractGoogleClientRequest[Void]`. So I also made it possible to explicitly pass the service enum here.